ecs-agent/netlib/model/ecscni/nsutil_linux_test.go

```diff
@@ -99,3 +101,32 @@ func TestBuildResolvConfig(t *testing.T) {
 	rst := nsUtil.BuildResolvConfig([]string{testNameServer}, []string{testSearchDomain})
 	require.Equal(t, fmt.Sprintf("nameserver %s\nsearch %s", testNameServer, testSearchDomain), rst)
 }
+
+// TestNewNSDuplicate verifies that creating two net ns will not fail.
+// It should ignore errors related to the default netns directory already existing.
+func TestNewNSDuplicate(t *testing.T) {
+	// Create a new instance of netnsutil.
+	ns := NewNetNSUtil()
+
+	// Create a temporary directory for testing.
+	tempDir, err := os.MkdirTemp("", tempNSPathPattern)
+	assert.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	// Default directory should not exist and check we can create with no issues.
+	defaultPath := filepath.Join(tempDir, "defaultns")
+	err = ns.NewNetNS(defaultPath)
+	assert.NoError(t, err)
+
+	// Newly created defaultNS should now exist.
+	_, err = os.Stat(defaultPath)
+	assert.NoError(t, err)
+
+	// Default directory already exists.
+	err = ns.NewNetNS(defaultPath)
+	assert.NoError(t, err)
+
+	// Newly created defaultNS should now exist.
+	_, err = os.Stat(defaultPath)
+	assert.NoError(t, err)
```

Comment:
nit: Does this check gain us anything? We already did this on L121-L123.

Reply:
If anything we should be checking for the new netns's path.

Reply:
Line 126 works to ensure the second call does not error if the path already exists.  (The condition we really care about in this new change).

And line 130, we're working to ensure it still exists and the second NewNetNs call did not have side-effects in this check.